app/components/autocomplete/autocomplete.tsx

```diff
@@ -0,0 +1,228 @@
+            borderColor: changeOpacity(theme.centerChannelColor, 0.2),
+            overflow: 'hidden',
+            borderRadius: 4,
+        },
+        hidden: {
+            display: 'none',
+        },
+        searchContainer: {
+            ...Platform.select({
+                android: {
+                    top: 42,
+                },
+                ios: {
+                    top: 55,
+                },
+            }),
+        },
+        shadow: {
+            shadowColor: '#000',
+            shadowOpacity: 0.12,
+            shadowRadius: 8,
+            shadowOffset: {
+                width: 0,
+                height: 8,
+            },
+        },
+    };
+});
+
+type Props = {
+    cursorPosition: number;
+    postInputTop: number;
+    rootId: string;
+    channelId: string;
+    isSearch?: boolean;
+    value: string;
+    enableDateSuggestion?: boolean;
+    isAppsEnabled: boolean;
+    offsetY?: number;
+    nestedScrollEnabled?: boolean;
+    updateValue: (v: string) => void;
+    hasFilesAttached: boolean;
+}
+
+const OFFSET_IPAD = 60;
+const AUTOCOMPLETE_MARGIN = 20;
+
+const Autocomplete = ({
+    cursorPosition,
+    postInputTop,
+    rootId,
+
+    //channelId,
+    isSearch = false,
+    value,
+
+    //enableDateSuggestion = false,
+    isAppsEnabled,
+    offsetY = 60,
+    nestedScrollEnabled = false,
+    updateValue,
+    hasFilesAttached,
+}: Props) => {
+    const theme = useTheme();
+    const isTablet = useIsTablet();
+    const style = getStyleFromTheme(theme);
+    const dimensions = useWindowDimensions();
+    const deviceHeight = dimensions.height;
+    const {defaultHeight: headerHeight} = useHeaderHeight(false, true, false);
+    const [keyboardHeight, setKeyboardHeight] = useState(0);
+
+    // const [showingAtMention, setShowingAtMention] = useState(false);
+    // const [showingChannelMention, setShowingChannelMention] = useState(false);
+    const [showingEmoji, setShowingEmoji] = useState(false);
+
+    // const [showingCommand, setShowingCommand] = useState(false);
+    // const [showingAppCommand, setShowingAppCommand] = useState(false);
+    // const [showingDate, setShowingDate] = useState(false);
+
+    const hasElements = showingEmoji; // || showingAtMention || showingChannelMention || showingCommand || showingAppCommand || showingDate;
+    const appsTakeOver = false; // showingAppCommand;
+
+    const maxListHeight = useMemo(() => {
+        if (Platform.OS === 'ios') {
+            const offset = isTablet ? OFFSET_IPAD : 0;
+            return Math.min(
+                deviceHeight - (keyboardHeight + headerHeight + AUTOCOMPLETE_MARGIN + offset),
+                (deviceHeight / 2) - (headerHeight + AUTOCOMPLETE_MARGIN + offset),
+            );
+        }
+
+        if (keyboardHeight) {
+            return (deviceHeight - (postInputTop + headerHeight + AUTOCOMPLETE_MARGIN));
+        }
+
+        return ((deviceHeight + headerHeight) / 2) - AUTOCOMPLETE_MARGIN;
```

Comment:
I may be wrong but I understand this logic to be "half of the available space minus the margin". Shouldn't "available space" be `deviceHeight - (headerHeight + postInputTop + AUTOCOMPLETE_MARGIN)`?

Reply:
Yes absolutely true, but it did not look very good taking almost all the screen, with this approach is large enough but still the channel below it is visible and gives a better vision in my opinion, do you disagree ?

Reply:
I agree. Just that `deviceHeight + headerHeight` is bigger than the actual screen. And we are saying to take half of that. Seems like something was overlooked when was being considered. What I would do is something like:
```
const availableSpace = deviceHeight - (postInputTop + HeaderHeight + AUTOCOMPLETE_MARGIN);
return availableSpace/2;
```
Does this make sense?

Reply:
it does, perhaps give it a try and see how does it look